tests/Avalonia.Markup.UnitTests/Data/ExpressionObserverTests_Indexer.cs

```diff
@@ -26,6 +26,36 @@ public class ExpressionObserverTests_Indexer
         }
 
         [Fact]
+        public async void Should_Get_UnsetValue_For_Invalid_Array_Index()
+        {
+            var data = new { Foo = new[] { "foo", "bar" } };
+            var target = new ExpressionObserver(data, "Foo[invalid]");
+            var result = await target.Take(1);
+
+            Assert.Equal(AvaloniaProperty.UnsetValue, result);
```

Comment:
Hmm, I think these should return a `BindingNotification` - that's a separate PR though :)